File: ServerApp/Entities/Periods/Year/YearStudent.cs
using System.Collections.Generic;
using Everest.AspNetStartup.Binding;
using Everest.AspNetStartup.Models;
using Microsoft.AspNetCore.Mvc;
using Newtonsoft.Json;

namespace Exam.Entities.Periods
{
    
    [ModelBinder(BinderType = typeof(ItemValueModelBinder))]
    public class YearStudent:Entity<long>
    {
        public virtual Student Student { get; set; }
        public long? StudentId { get; set; }

        [JsonIgnore]
        public virtual YearLevel YearLevel { get; set; }
        public long YearLevelId { get; set; }

        [JsonIgnore]
        public virtual YearLevelSpeciality YearLevelSpeciality { get; set; }
        public long? YearLevelSpecialityId { get; set; }

        [JsonIgnore]
        public virtual List<SemesterStudent> SemesterStudents { get; set; }
    }
}

File: webapp/apps/regolia/src/models/entities/year/year-student.ts
import {Entity} from "../entity";
import {YearLevel} from "./year-level";
import {Student} from "../member";
import {YearLevelSpeciality} from "./year-level-speciality";
import {SemesterStudent} from "../semester";

export class YearStudent extends Entity<number> {

  constructor(value: any = {}) {
    super();

    this.id = value.id;
    this.registrationDate = new Date(value.registrationDate);

    this.yearLevelId = value.yearLevelId;
    this.studentId = value.studentId;
    this.student = new Student(value.student);
    this.yearLevelSpecialityId = value.yearLevelSpecialityId;
  }

  yearLevelId: number;
  yearLevel: YearLevel;

  student: Student;
  studentId: number;

  yearLevelSpeciality: YearLevelSpeciality;
  yearLevelSpecialityId: number;

  isNew: boolean = false;
  semesterStudents: SemesterStudent[];

  url(path?: string): string {
    const url = `${this.yearLevel?.url()}/students/${this.id}`;
    if(path) {
      return `${url}/${path}`;
    }
    return url;
  }
}
